### PR TITLE
[Modlog] add timestamp to [p]casesfor and [p]case

### LINKED ISSUE #4118: Feature request, core modlog.py, Add Timestamp for sanitized embeds.
# Feature request

Add Timestamp for sanitization outside embeds with core modlog python file.

#### Select the type of feature you are requesting:

- [x] Cog
- [ ] Command
- [ ] API functionality

#### Describe your requested feature

Referencing code section: https://github.com/Cog-Creators/Red-DiscordBot/blob/28b5be21b3d3dac5313c2c701bdb4ce491889b04/redbot/core/modlog.py#L380-L394

We should have the timestamp being shown on the sanitized embed of the current case as seen below

```python
        else:
            user = filter_mass_mentions(filter_urls(user))  # Further sanitization outside embeds
            case_text = ""
            case_text += "{}\n".format(title)
            case_text += _("**User:** {}\n").format(user)
            case_text += _("**Moderator:** {}\n").format(moderator)
            case_text += _("**Time Stamp:** {}\n").format(datetime.fromtimestamp(self.created_at)) # Time Stamp added here
            case_text += "{}\n".format(reason)
            if until and duration:
                case_text += _("**Until:** {}\n**Duration:** {}\n").format(until, duration)
            if self.channel:
                case_text += _("**Channel**: {}\n").format(self.channel.name)
            if amended_by:
                case_text += _("**Amended by:** {}\n").format(amended_by)
            if last_modified:
                case_text += _("**Last modified at:** {}\n").format(last_modified)
            return case_text.strip()
```

Result: 
image

### PR DESCRIPTION
**Pull request by Dav-Git**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Added timestamps display to the non-embedded version of ``[p]case`` and ``[p]casesfor``.
This closes #4118 
grafik
grafik

### CODE REVIEW

**Comment by Dav-Git** [comment]
ignore latest empty commit, i'm dumb

File: redbot/cogs/modlog/modlog.py [hunk 1]
```diff
@@ -112,7 +112,13 @@ async def case(self, ctx: commands.Context, number: int):
             if await ctx.embed_requested():
                 await ctx.send(embed=await case.message_content(embed=True))
             else:
-                await ctx.send(await case.message_content(embed=False))
+                message = _("{case}\n**Timestamp**: {timestamp}").format(
```
Comment (Jackenmen) [inline redbot/cogs/modlog/modlog.py]: The colon is bolder for other fields, so it should be bolded here too for consistency.
```suggestion
                message = _("{case}\n**Timestamp:** {timestamp}").format(
```

File: redbot/cogs/modlog/modlog.py [hunk 2]
```diff
@@ -138,8 +144,18 @@ async def casesfor(self, ctx: commands.Context, *, member: Union[discord.Member,
...
 
         embed_requested = await ctx.embed_requested()
-
-        rendered_cases = [await case.message_content(embed=embed_requested) for case in cases]
+        if embed_requested:
+            rendered_cases = [await case.message_content(embed=True) for case in cases]
+        elif not embed_requested:
+            rendered_cases = []
+            for case in cases:
+                message = _("{case}\n**Timestamp**: {timestamp}").format(
```
Comment (Jackenmen) [inline redbot/cogs/modlog/modlog.py]: The colon is bolder for other fields, so it should be bolded here too for consistency.
```suggestion
                message = _("{case}\n**Timestamp:** {timestamp}").format(
```

**Comment by Dav-Git** [comment]
I just accepted incoming here, let's hope datetime isn't needed anywhere I guess

**Comment by Dav-Git** [comment]
@jack1142 does any of this now require different handling with #4017 merged? I've honestly not quite kept up with the underlying issue there and am unsure if this sort of display is affected.

File: redbot/cogs/modlog/modlog.py [hunk 3]
```diff
@@ -117,7 +117,13 @@ async def case(self, ctx: commands.Context, number: int):
             if await ctx.embed_requested():
                 await ctx.send(embed=await case.message_content(embed=True))
             else:
-                await ctx.send(await case.message_content(embed=False))
+                message = _("{case}\n**Timestamp**: {timestamp}").format(
+                    case=await case.message_content(embed=False),
+                    timestamp=datetime.fromtimestamp(case.created_at).strftime(
+                        "%Y-%m-%d %H:%M:%S"
```
Comment (Jackenmen) [inline redbot/cogs/modlog/modlog.py]: Updated handling per #4017:
```suggestion
                    timestamp=datetime.utcfromtimestamp(case.created_at).strftime(
                        "%Y-%m-%d %H:%M:%S UTC"
```

File: redbot/cogs/modlog/modlog.py [hunk 4]
```diff
@@ -143,8 +149,18 @@ async def casesfor(self, ctx: commands.Context, *, member: Union[discord.Member,
...
-        rendered_cases = [await case.message_content(embed=embed_requested) for case in cases]
+        if embed_requested:
+            rendered_cases = [await case.message_content(embed=True) for case in cases]
+        elif not embed_requested:
+            rendered_cases = []
+            for case in cases:
+                message = _("{case}\n**Timestamp**: {timestamp}").format(
+                    case=await case.message_content(embed=False),
+                    timestamp=datetime.fromtimestamp(case.created_at).strftime(
+                        "%Y-%m-%d %H:%M:%S"
```
Comment (Jackenmen) [inline redbot/cogs/modlog/modlog.py]: Updated handling per #4017:
```suggestion
                    timestamp=datetime.utcfromtimestamp(case.created_at).strftime(
                        "%Y-%m-%d %H:%M:%S UTC"
```

**Review by Jackenmen — CHANGES_REQUESTED** [review]

**Review by Jackenmen — APPROVED** [review]
LGTM, thanks for PRing.

**Pull request merged**